aodncore/pipeline/files.py

```diff
@@ -244,8 +247,20 @@ def __init__(self, local_path, name=None, archive_path=None, dest_path=None, is_
         self._is_harvest_undone = False
         self._is_upload_undone = False
 
-        self._check_result = None
-        self._mime_type = None
+        # attributes which must be assigned by the property setter for validation. The backing variable is intentionally
+        # initialised to a safe default, before the setter is called if the calling code has supplied a value for the
+        # corresponding parameters
+        self._file_update_callback = None
+        if file_update_callback is not None:
+            self.file_update_callback = file_update_callback
+
+        self._check_type = PipelineFileCheckType.UNSET
+        if check_type is not None:
+            self.check_type = check_type
+
+        self._publish_type = PipelineFilePublishType.UNSET
+        if publish_type is not None:
+            self.publish_type = publish_type
 
     @classmethod
     def from_remotepipelinefile(cls, remotepipelinefile, is_deletion=False):
```

Comment:
It would be helpful to add all the `__init__` ags here (or at least `publish_type` and `late_deletion`) as optional arguments here. A common pattern would be use the results of a WFS query to find old files, then add them as deletions (see e.g. [soop_trv](https://github.com/aodn/python-aodndata/blob/8de9b2bb20502ccea8407cd8f59ed87525969ce4/aodndata/soop/soop_trv.py#L22)).
That could then become
```python
        files_to_delete = self.state_query.query_wfs_urls_for_layer('soop_trv_duplicate_url')
        for f in files_to_delete:
            file_to_delete = PipelineFile.from_remotepipelinefile(f, is_deletion=True,
                                                                  publish_type=PipelineFilePublishType.DELETE_UNHARVEST)
            self.add_pipelinefile(file_to_delete)
```

Reply:
Done. It is reasonable that the alternate constructor can take the same parameters as the primary one.